src/App.tsx

```diff
@@ -1,61 +1,137 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const getTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+const getNewTodoId = (todos: Todo[]) => {
+  const todoIds = todos.map(todo => todo.id);
+  const maxId = Math.max(...todoIds);
+
+  return Number.isFinite(maxId)
+    ? maxId + 1
+    : 1;
+};
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(getTodos);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [isTitleValid, setIsTitleValid] = useState(false);
+  const [isUserValid, setIsUserValid] = useState(false);
+
+  const handleAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!userId) {
+      setIsUserValid(true);
+    }
+
+    if (!title) {
+      setIsTitleValid(true);
+    }
+
+    if (!userId || !title) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getNewTodoId(todos),
+      completed: false,
+      user: getUserById(Number(userId)),
+      title: title.trim(),
+      userId: Number(userId),
+    };
+
+    setTodos([...todos, newTodo]);
+    setTitle('');
+    setUserId(0);
+  };
+
+  const changeTitle = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setIsTitleValid(false);
+    setTitle(event.target.value);
+  };
```

Comment:
I can push todo with spaces in title

![image](https://github.com/mate-academy/react_add-todo-form/assets/84296041/66c65acd-4a80-4cce-83dd-8f0e27fb1185)